### PR TITLE
Add Flotiq integration

### PR DESCRIPTION
**Pull request by trzcina**
#### Description

This MR introduces documentation for integrating [Flotiq](https://flotiq.com) headless CMS with Astro, offering developers a guide to enriching their Astro projects with dynamic content from Flotiq. 

It covers prerequisites, setting up content types in Flotiq, and connecting Astro components to Flotiq data, complete with code snippets and steps to display blog posts fetched from Flotiq.

### CODE REVIEW

**Comment by trzcina** [comment]
Hello, could you please take a look at this MR when you have a moment? Thank you! :)

File: src/content/docs/en/guides/cms/flotiq.mdx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Run the project with the `astro dev` command; this will allow us to see changes and possible messages in real-time.
+
+Go to the `/blog` page, e.g. `http://localhost:4322/blog`
+
+*We'll modify the component that lists posts* to showcase content from Flotiq, altering the data retrieval method by utilizing the endpoint /api/v1/content/blogpost. 
+Remember to include your token in the request header.
+
+```diff title="src/pages/blog/index.astro"
+import FormattedDate from '../../components/FormattedDate.astro';
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/cms/flotiq.mdx:59]: We have a custom syntax we can use for these
``````suggestion
```astro title="src/pages/blog/index.astro" ins={7-17} del={4-6}
---
import FormattedDate from '../../components/FormattedDate.astro';
``````

File: src/content/docs/en/guides/cms/flotiq.mdx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
++  },
++})
++  .then(response => response.json())
++  .then(response => {
++    console.debug({responseFromFlotiq: response});
++    return response.data;
++  });
+--- 
+
+<!doctype html>
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/cms/flotiq.mdx:77]:
```suggestion
```

File: src/content/docs/en/guides/cms/flotiq.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+```
+ 
+Let's adjust the template of a single post to the structure obtained from Flotiq.
+
+```diff title="src/pages/blog/index.astro"
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/cms/flotiq.mdx:82]:
``````suggestion
```astro title="src/pages/blog/index.astro" ins={6-7, 10} del={4-5, 9}
``````

**Review by TheOtterlord — COMMENTED** [review]
Thanks for the PR @trzcina! Sorry for the delay. We're catching up on PRs and issues at the moment.

I've followed the guide, and have some feedback below. However, I feel it's missing a section at the end. Our CMS guides usually explain how to render the content pulled from the CMS, but this seems to hint at an SDK (with no link) that you could to use to render pages? It would be great to have an example of rendering pages in the guide if possible.

I've included the rest of my initial review below, but the above is the most important, as it doesn't yet feel like a complete guide.

Let me know if you have any questions/comments!

File: src/content/docs/en/guides/cms/flotiq.mdx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### Next steps
+
+Next Steps:
+
+Expand other pages, such as individual blog posts, to fetch data from Flotiq. This involves adapting the data retrieval logic to display detailed views for each post.
+
+Refactor the communication with Flotiq into a separate service. By isolating the API calls, your project becomes more maintainable and scalable, making it easier to manage the integration points.
+
+Utilize the Flotiq SDK for API interactions. The SDK simplifies working with the Flotiq API by providing convenient methods for fetching and managing content, enhancing the development experience and speeding up the integration process.
```
[the remark this answers is not in the document]
Comment (trzcina) [reply src/content/docs/en/guides/cms/flotiq.mdx:89]: The SDK is another method of communicating with Flotiq, described here https://flotiq.com/docs/API/generate-package/. In the tutorial we described a simple "fetch" method, sufficient for many applications. We just wanted to point out that you can also use the SDK, especially in larger projects.

**Comment by trzcina** [comment]
Hi @TheOtterlord. Thanks for the review! I've applied your suggestions and replied to your comment. I'd appreciate it if you could check it again.

**Comment by TheOtterlord** [comment]
Thanks! I'm out of town at the moment, but I'll be back on Saturday to review the new version.

**Review by TheOtterlord — COMMENTED** [review]
The first section looks great! I still feel the tutorial is missing any way to render the complete content of a post. Looking at the output of fetch, this specifically doesn't seem trivial to render, but if there is a way and I'm missing it, that would make this guide much more helpful.

**Comment by sarah11918** [comment]
Hi, it has been a few weeks and there has been no response to the request for more details about fetching data.

And, the example provided has a lot going on here that should not be in a minimal example. I don't think this makes it obvious how to use Flotiq and instead makes this difficult to follow: there's an extra, unnecessary date component, posts is defined twice, once fetching from content collections and once fetching from Flotiq, there's a debug statement.

This doesn't meet our standards for a helpful guide, and especially with no recent response from the PR reviewer, I'll be closing this PR.

**Pull request closed without merging**